Homework: Ticket Services Hierarchy

HKV203 opened the pull request.

Commit:
Homework: Ticket Service Hierarchy

VolgaDaniluk commented on src/concerttickets/TicketService.java:
```diff
@@ -1,9 +1,36 @@
+/**
+ * clone + Cloneable
+ * <p>
+ * static count++ for ID
+ */
+
```
what is it?

HKV203 replied:
These were my own TODO comments that I forgot to delete

VolgaDaniluk commented on src/concerttickets/models/tickets/Ticket.java:
```diff
@@ -0,0 +1,220 @@
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+//import java.util.Random;
```
remove , please

VolgaDaniluk commented on src/concerttickets/models/tickets/Ticket.java:
```diff
@@ -0,0 +1,220 @@
+
+    }
+
+    /*
+    private int generateId() {
+        StringBuilder newId = new StringBuilder();
+        Random random = new Random();
+        String characters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        for (int i = 0; i < 4; i++) {
+            newId.append(characters.charAt(random.nextInt(characters.length())));
+        }
+        return newId.toString();
+    }
+    */
```
remove, please

VolgaDaniluk commented on src/concerttickets/models/tickets/Ticket.java:
```diff
@@ -0,0 +1,220 @@
+    }
+    */
+    private void createLimitedTicket(String hall, int code, LocalDateTime time) {
+        this.setId(Ticket.lastId++);
```
why does the id start from 1000?

HKV203 replied:
I probably misunderstood one of the previous home tasks with four digits or letters ID

VolgaDaniluk commented on src/concerttickets/models/tickets/Ticket.java:
```diff
@@ -0,0 +1,220 @@
+        return this.sector;
+    }
+
+    public BigDecimal maxWeight() {
```
'getMaxWeight'
otherwise it's not clear what does this method do

VolgaDaniluk reviewed (commented):
equals, hashCode and toString don't overrided in all necessary classes

Mopgonpblr commented:
> equals, hashCode and toString don't overrided in all necessary classes

The home task says "Override equals(), toString() and hashCode() in your Ticket class..."

Commit:
Fixing comment sections, the ID field in Ticket now starts from 0 and maxWeight() method is now called getMaxWeight()

Commit:
Methods equals(), hashCode() and toString() are overridden in more classes

Commit:
2. Added printDefault() method that uses the default implementation of print() method (not the overriden one)
4. Different implementations of printRole() are called in method main() to show the polymorphism

Mopgonpblr commented:
2. Added printDefault() method that uses the default implementation of print() method (not the overridden one)

4. Different implementations of printRole() are called in method main() to show the polymorphism

6. equals(), hashCode() and toString() methods are overridden in more classes
